examples/uma-server/vasp2.go

```diff
@@ -486,6 +504,204 @@ func (v *Vasp2) handleUtxoCallback(context *gin.Context) {
 	context.Status(http.StatusOK)
 }
 
+func (v *Vasp2) handleCreateInvoice(context *gin.Context) {
+	invoice, err := v.createInvoice(context, false)
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{
+			"status": "ERROR",
+			"reason": fmt.Sprintf("Failed to create invoice: %v", err),
+		})
+		return
+	}
+
+	invoiceString, err := invoice.ToBech32String()
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{
+			"status": "ERROR",
+			"reason": fmt.Sprintf("Failed to convert invoice to bech32 string: %v", err),
+		})
+		return
+	}
+
+	context.JSON(http.StatusOK, invoiceString)
+}
+
+func (v *Vasp2) handleCreateAndSendInvoice(context *gin.Context) {
+	invoice, err := v.createInvoice(context, true)
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{
+			"status": "ERROR",
+			"reason": fmt.Sprintf("Failed to create invoice: %v", err),
+		})
+		return
+	}
+
+	// Send the invoice to the sender.
+
+	// Step 1: Get the sender's domain from sender's UMA address.
+	senderUma := *invoice.SenderUma
+	senderVaspDomain, err := uma.GetVaspDomainFromUmaAddress(senderUma)
+	if strings.Contains(senderVaspDomain, "local") {
+		senderVaspDomain = "http://" + senderVaspDomain
+	} else {
+		senderVaspDomain = "https://" + senderVaspDomain
+	}
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, gin.H{
+			"status": "ERROR",
+			"reason": fmt.Sprintf("Failed to get sender's VASP domain: %v", err),
+		})
+		return
+	}
+
+	// Step 2: Query sender's domain /.well-known/uma-configruration to get sender's request URL.
+	// Make a GET request to the sender's /.well-known/uma-configuration endpoint.
+	url := senderVaspDomain + "/.well-known/uma-configuration"
+
+	resp, err := http.Get(url)
+    if err != nil {
+        fmt.Println("Error making GET request:", err)
+        return
+    }
+	defer resp.Body.Close()
+
+	    type UmaConfig struct {
+        UmaRequestEndpoint string `json:"uma_request_endpoint"`
+    }
+
+    var config UmaConfig
+
+	// Read the response body
+	body, err := io.ReadAll(resp.Body)
+    if err != nil {
+        fmt.Println("Error reading response body:", err)
+        return
+    }
+
+    // Parse the JSON response
+    err = json.Unmarshal(body, &config)
+    if err != nil {
+        fmt.Println("Error parsing JSON response:", err)
+        return
+    }
+
+	requestEndpoint := config.UmaRequestEndpoint
```

Comment:
Here too... something is messed up with indents in this function generally. Not sure what happened.